sympy/functions/elementary/hyperbolic.py

```diff
@@ -287,19 +287,19 @@ def _eval_is_positive(self):
         re, im = arg.as_real_imag()
         im_mod = im % (2*pi)
 
-        if im_mod == 0:
+        if im_mod.is_zero:
             return True
 
-        if re == 0:
+        if re.is_zero:
             if im_mod < pi/2 or im_mod > 3*pi/2:
                 return True
             elif im_mod >= pi/2 or im_mod <= 3*pi/2:
                 return False
 
-        return fuzzy_or([fuzzy_and([fuzzy_bool(Eq(re, 0)),
-                         fuzzy_or([fuzzy_bool(im_mod < pi/2),
```

Comment:
[This change](https://github.com/sympy/sympy/pull/17846/files#r341847641) helps avoiding `fuzzy_bool` repeatedly, for non-`bool` types.
If there are any better(other than the one on right and in [this comment]([This change](https://github.com/sympy/sympy/pull/17846/files#r341847641) helps avoiding `fuzzy_bool` repeatedly, for non-`bool` types.)) suggestions for implementing this more compactly, then please let me know.